extra "/" in DAV URL makes server return "500 Internal Server Error"

### Steps to reproduce
1. access webDAV endpoint with an extra `/` before `remote.php` e.g `curl -u admin:admin  "http://localhost/owncloud-core//remote.php/dav/files/admin/welcome.txt"  -v`

### Expected behaviour
nice error with some 4xx code returned

### Actual behaviour
500 Internal Server Error

### Server configuration
**Operating system**:
Ubuntu
**Web server:**
Apache2
**Database:**
SQLite
**PHP version:**
7.1.26
**ownCloud version:** (see ownCloud admin page)
11.0.0 prealpha (git) 
**Updated from an older ownCloud or fresh install:**
fresh
**Where did you install ownCloud from:**
git

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "oce1t8r2vis9",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "172.17.0.1"
        ],
        "datadirectory": "\/home\/artur\/www\/owncloud-core\/data",
        "overwrite.cli.url": "http:\/\/localhost\/owncloud-core",
        "dbtype": "sqlite3",
        "version": "11.0.0.0",
        "logtimezone": "UTC",
        "htaccess.RewriteBase": "\/owncloud-core\/",
        "ldapIgnoreNamingRules": false,
        "enable_previews": false,
        "csrf.disabled": true,
        "loglevel": 0,
        "theme": "",
        "license-key": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "1025",
        "sharing.federation.allowHttpFallback": true,
        "files_external_allow_create_new_local": "true",
        "skeletondirectory": "\/home\/artur\/www\/owncloud-core\/tests\/acceptance\/..\/..\/apps\/testing\/data\/apiSkeleton",
        "log_type": "owncloud",
        "firewall.debug": 1,
        "maintenance": false,
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "dav.enable.async": true
    }
}
```

**List of activated apps:**

```
Enabled:
  - activity: 2.4.1
  - comments: 0.3.0
  - dav: 0.4.0
  - diagnostics: 0.1.3
  - enterprise_key: 0.1.4
  - federatedfilesharing: 0.3.1
  - federation: 0.1.0
  - files: 1.5.2
  - files_external: 0.7.1
  - files_sharing: 0.11.0
  - files_texteditor: 2.2.1
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - firewall: 2.7.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - systemtags_management: 0.3.1
  - testing: 0.1.0
  - user_management: 0.0.1
  - windows_network_drive: 0.7.2
  - workflow: 0.3.0
Disabled:
  - brute_force_protection
  - files_antivirus
  - files_classifier
  - files_primary_s3
  - guests
  - multidirtest
  - notifications
  - oauth2
  - password_policy
  - ransomware_protection
  - search_elastic
  - twofactor_totp
  - updatenotification
  - user_ldap

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
### Logs
#### ownCloud log (data/owncloud.log)
```{"reqId":"rIW2E7vfRsC5tszMac8O","level":3,"time":"2019-02-04T07:40:28+00:00","remoteAddr":"127.0.0.1","user":"--","app":"remote","method":"GET","url":"\/owncloud-core\/\/remote.php\/dav\/files\/admin\/welcome.txt","message":"Exception: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\Forbidden\",\"Message\":\"Requested uri (\\\/owncloud-core\\\/remote.php\\\/dav\\\/files\\\/admin\\\/welcome.txt) is out of base uri (\\\/owncloud-core\\\/\\\/remote.php\\\/dav\\\/)\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/artur\\\/www\\\/owncloud-core\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(534): Sabre\\\\DAV\\\\Server->calculateUri('\\\/owncloud-core\\\/...')\\n#1 \\\/home\\\/artur\\\/www\\\/owncloud-core\\\/apps\\\/dav\\\/lib\\\/Server.php(309): Sabre\\\\DAV\\\\Server->getRequestUri()\\n#2 \\\/home\\\/artur\\\/www\\\/owncloud-core\\\/apps\\\/dav\\\/lib\\\/Server.php(148): OCA\\\\DAV\\\\Server->isRequestForSubtree(Array)\\n#3 \\\/home\\\/artur\\\/www\\\/owncloud-core\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(30): OCA\\\\DAV\\\\Server->__construct(Object(OC\\\\AppFramework\\\\Http\\\\Request), '\\\/owncloud-core\\\/...')\\n#4 \\\/home\\\/artur\\\/www\\\/owncloud-core\\\/remote.php(175): require_once('\\\/home\\\/artur\\\/www...')\\n#5 {main}\",\"File\":\"\\\/home\\\/artur\\\/www\\\/owncloud-core\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php\",\"Line\":568}"}
```

@individual-it the issue dosn't exist in current master
I tryed to do git bisect i cannot build core on the old version, with error
```
Building core

Note: You can type 'make help' for more targets

php build/composer.phar install --dev

                                                                                                                       
  [RuntimeException]                                                                                                   
  require.icewind/Streams is invalid, it should not contain uppercase characters. Please use icewind/streams instead.  
                                                                                                                       

make: *** [Makefile:134: lib/composer/phpunit] Error 1
```

I'll try adding test for this so we dont get regressions later

> 
> 
> @individual-it the issue dosn't exist in current master
> I tryed to do git bisect i cannot build core on the old version, with error
> 
> ```
> Building core
> 
> Note: You can type 'make help' for more targets
> 
> php build/composer.phar install --dev
> 
>                                                                                                                        
>   [RuntimeException]                                                                                                   
>   require.icewind/Streams is invalid, it should not contain uppercase characters. Please use icewind/streams instead.  
>                                                                                                                        
> 
> make: *** [Makefile:134: lib/composer/phpunit] Error 1
> ```
> 
> I'll try adding test for this so we dont get regressions later

How did you test this, since in the bug says is version 11.0.0 alpha, with php 7.1 and first, I can not find version 11 and php 7.1 is not supported anymore in version 10.5

--- a/tests/acceptance/features/apiAuthWebDav/webDavSpecialURLs.feature
+++ b/tests/acceptance/features/apiAuthWebDav/webDavSpecialURLs.feature
@@ -1,0 +1,110 @@
+@api @skipOnOcV10
+Feature: make webdav request with special urls
+
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And user "Alice" has uploaded file with content "some data" to "/textfile0.txt"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And user "Alice" has created folder "/PARENT"
+    And user "Alice" has created folder "/FOLDER"
+    And user "Alice" has uploaded file with content "some data" to "/PARENT/parent.txt"
+
+
+  Scenario: send DELETE requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "DELETE" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "204"
+
+
+  Scenario: send GET requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "GET" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "200"
+
+
+  Scenario: send LOCK requests to webDav endpoints with 2 slashes
+    When the user "Alice" requests these endpoints with "LOCK" to get property "d:shared" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "200"
+
+
+  Scenario: send MKCOL requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "MKCOL" using password "%regular%" about user "Alice"
+      | endpoint                                   |
+      | //remote.php/webdav/PARENT1                |
+      | /remote.php//webdav/PARENT2                |
+      | //remote.php//webdav/PARENT3               |
+      | //remote.php/dav//files/%username%/PARENT4 |
+      | /remote.php/dav/files/%username%//PARENT5  |
+      | /remote.php/dav//files/%username%/PARENT6  |
+    Then the HTTP status code of responses on all endpoints should be "201"
+
+
+  Scenario: send MOVE requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "MOVE" using password "%regular%" about user "Alice"
+      | endpoint                                             | destination                                          |
+      | //remote.php/webdav/textfile0.txt                    | /remote.php/webdav/textfileZero.txt                  |
+      | /remote.php//dav/files/%username%/textfile1.txt      | /remote.php/dav/files/%username%/textfileOne.txt     |
+      | /remote.php/webdav//PARENT                           | /remote.php/webdav/PARENT1                           |
+      | //remote.php/dav/files/%username%//PARENT1           | /remote.php/dav/files/%username%/PARENT2             |
+      | /remote.php/dav//files/%username%/PARENT2/parent.txt | /remote.php/dav/files/%username%/PARENT2/parent1.txt |
+    Then the HTTP status code of responses on all endpoints should be "201"
+
+
+  Scenario: send POST requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "POST" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "500" or "501"
+
+
+  Scenario: send PROPFIND requests to webDav endpoints with 2 slashes
+    When the user "Alice" requests these endpoints with "PROPFIND" to get property "d:href" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "207"
+
+
+  Scenario: send PROPPATCH requests to webDav endpoints with 2 slashes
+    When the user "Alice" requests these endpoints with "PROPPATCH" to set property "d:getlastmodified" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "207"
+
+
+  Scenario: send PUT requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "PUT" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                             |
+      | //remote.php/webdav/textfile0.txt                    |
+      | /remote.php//webdav/textfile1.txt                    |
+      | //remote.php//dav/files/%username%/textfile1.txt     |
+      | /remote.php/dav/files/%username%/textfile7.txt       |
+      | //remote.php/dav/files/%username%/PARENT//parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "204" or "201"

--- a/tests/acceptance/features/apiAuthWebDav/webdavMultiShashOC10Issue34365.feature
+++ b/tests/acceptance/features/apiAuthWebDav/webdavMultiShashOC10Issue34365.feature
@@ -1,0 +1,139 @@
+@api @notToImplementOnOCIS
+Feature: current oC10 behavior for issue-34365
+  This is the current behaviour in owncloud10
+  After issue #34365 is fixed delete this file and un-skip the tests from apiAuthWebDav/webDavSpecialURLs.feature
+
+  Background:
+    Given user "Alice" has been created with default attributes and without skeleton files
+    And user "Alice" has uploaded file with content "some data" to "/textfile0.txt"
+    And user "Alice" has uploaded file with content "some data" to "/textfile1.txt"
+    And user "Alice" has created folder "/PARENT"
+    And user "Alice" has created folder "/FOLDER"
+    And user "Alice" has uploaded file with content "some data" to "/PARENT/parent.txt"
+
+
+  Scenario: send DELETE requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "DELETE" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+    Then the HTTP status code of responses on all endpoints should be "204"
+    When user "Alice" requests these endpoints with "DELETE" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "500"
+
+
+  Scenario: send GET requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "GET" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+    Then the HTTP status code of responses on all endpoints should be "200"
+    When user "Alice" requests these endpoints with "GET" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "500"
+
+
+  Scenario: send LOCK requests to webDav endpoints with 2 slashes
+    When the user "Alice" requests these endpoints with "LOCK" to get property "d:shared" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+    Then the HTTP status code of responses on all endpoints should be "200"
+    When the user "Alice" requests these endpoints with "LOCK" to get property "d:shared" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "500"
+
+
+  Scenario: send MKCOL requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "MKCOL" using password "%regular%" about user "Alice"
+      | endpoint                                   |
+      | /remote.php//webdav/PARENT2                |
+      | /remote.php/dav/files/%username%//PARENT5  |
+      | /remote.php/dav//files/%username%/PARENT6  |
+    Then the HTTP status code of responses on all endpoints should be "201"
+    When user "Alice" requests these endpoints with "MKCOL" using password "%regular%" about user "Alice"
+      | endpoint                                   |
+      | //remote.php/webdav/PARENT1                |
+      | //remote.php//webdav/PARENT3               |
+      | //remote.php/dav//files/%username%/PARENT4 |
+    Then the HTTP status code of responses on all endpoints should be "500"
+
+
+  Scenario: send MOVE requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "MOVE" using password "%regular%" about user "Alice"
+      | endpoint                                             | destination                                          |
+      | /remote.php//dav/files/%username%/textfile1.txt      | /remote.php/dav/files/%username%/textfileOne.txt     |
+      | /remote.php/webdav//PARENT                           | /remote.php/webdav/PARENT1                           |
+      | /remote.php/dav//files/%username%/PARENT1/parent.txt | /remote.php/dav/files/%username%/PARENT1/parent1.txt |
+    Then the HTTP status code of responses on all endpoints should be "201"
+    When user "Alice" requests these endpoints with "MOVE" using password "%regular%" about user "Alice"
+      | endpoint                                             | destination                                          |
+      | //remote.php/webdav/textfile0.txt                    | /remote.php/webdav/textfileZero.txt                  |
+      | //remote.php/dav/files/%username%//PARENT1           | /remote.php/dav/files/%username%/PARENT2             |
+    Then the HTTP status code of responses on all endpoints should be "500"
+
+
+  Scenario: send POST requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "POST" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | /remote.php//webdav/PARENT                          |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "501"
+    When user "Alice" requests these endpoints with "POST" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "500"
+
+
+  Scenario: send PROPFIND requests to webDav endpoints with 2 slashes
+    When the user "Alice" requests these endpoints with "PROPFIND" to get property "d:href" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+    Then the HTTP status code of responses on all endpoints should be "207"
+    When the user "Alice" requests these endpoints with "PROPFIND" to get property "d:href" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "500"
+
+
+  Scenario: send PROPPATCH requests to webDav endpoints with 2 slashes
+    When the user "Alice" requests these endpoints with "PROPPATCH" to set property "d:getlastmodified" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | /remote.php//dav/files/%username%/PARENT/parent.txt |
+      | /remote.php//webdav/PARENT                          |
+    Then the HTTP status code of responses on all endpoints should be "207"
+    When the user "Alice" requests these endpoints with "PROPPATCH" to set property "d:getlastmodified" with password "%regular%" about user "Alice"
+      | endpoint                                            |
+      | //remote.php/webdav/textfile0.txt                   |
+      | //remote.php//dav/files/%username%/textfile1.txt    |
+      | //remote.php/dav//files/%username%//FOLDER           |
+    Then the HTTP status code of responses on all endpoints should be "500"
+
+
+  Scenario: send PUT requests to webDav endpoints with 2 slashes
+    When user "Alice" requests these endpoints with "PUT" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                             |
+      | /remote.php//webdav/textfile1.txt                    |
+      | /remote.php/dav/files/%username%/textfile7.txt       |
+    Then the HTTP status code of responses on all endpoints should be "204" or "201"
+    When user "Alice" requests these endpoints with "PUT" including body "doesnotmatter" using password "%regular%" about user "Alice"
+      | endpoint                                             |
+      | //remote.php/webdav/textfile0.txt                    |
+      | //remote.php//dav/files/%username%/textfile1.txt     |
+      | //remote.php/dav/files/%username%/PARENT//parent.txt |
+    Then the HTTP status code of responses on all endpoints should be "500"

--- a/tests/acceptance/features/bootstrap/AuthContext.php
+++ b/tests/acceptance/features/bootstrap/AuthContext.php
@@ -288,6 +288,26 @@ class AuthContext implements Context {
 	}
 
 	/**
+	 * @When /^the user "([^"]*)" requests these endpoints with "([^"]*)" to (?:get|set) property "([^"]*)" with password "([^"]*)" about user "([^"]*)"$/
+	 *
+	 * @param string $user
+	 * @param string $method
+	 * @param string $property
+	 * @param string $ofUser
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userRequestsEndpointsWithPasswordWithProperty(
+		$user, $method, $property, $ofUser, TableNode $table
+	) {
+		$this->userRequestsEndpointsWithPassword(
+			$user, $method, $ofUser, $table, $property
+		);
+	}
+
+	/**
 	 * @When the user :user requests these endpoints with :method with body :body using basic auth and generated app password about user :ofUser
 	 *
 	 * @param string $user
@@ -360,17 +380,22 @@ class AuthContext implements Context {
 	 * @param string $method
 	 * @param string $password
 	 * @param TableNode $table
+	 * @param string $property
 	 *
 	 * @return void
 	 * @throws Exception
 	 */
-	public function userRequestsEndpointsWithPassword($user, $method, $password, TableNode $table) {
+	public function userRequestsEndpointsWithPassword($user, $method, $password, TableNode $table, $property = null) {
 		$user = $this->featureContext->getActualUsername($user);
 		$this->featureContext->emptyLastOCSStatusCodesArray();
 		$this->featureContext->emptyLastHTTPStatusCodesArray();
 		$this->featureContext->verifyTableNodeColumns($table, ['endpoint']);
 		foreach ($table->getHash() as $row) {
-			$this->userRequestsURLWithUsingBasicAuth($user, $row['endpoint'], $method, $password);
+			$body = null;
+			if ($property !== null) {
+				$body = $this->featureContext->getBodyForOCSRequest($method, $property);
+			}
+			$this->userRequestsURLWithUsingBasicAuth($user, $row['endpoint'], $method, $password, $body);
 			$this->featureContext->pushToLastHttpStatusCodesArray(
 				$this->featureContext->getResponse()->getStatusCode()
 			);
@@ -750,11 +775,12 @@ class AuthContext implements Context {
 	 * @param string $method
 	 * @param string $password
 	 * @param string $body
+	 * @param Array $header
 	 *
 	 * @return void
 	 * @throws Exception
 	 */
-	public function userRequestsURLWithUsingBasicAuth($user, $url, $method, $password = null, $body = null) {
+	public function userRequestsURLWithUsingBasicAuth($user, $url, $method, $password = null, $body = null, $header = null) {
 		$userRenamed = $this->featureContext->getActualUsername($user);
 		$url = $this->featureContext->substituteInLineCodes(
 			$url, $userRenamed
@@ -762,10 +788,10 @@ class AuthContext implements Context {
 		if ($password === null) {
 			$authString = "$userRenamed:" . $this->featureContext->getPasswordForUser($user);
 		} else {
-			$authString = $userRenamed . ":" . $password;
+			$authString = $userRenamed . ":" . $this->featureContext->getActualPassword($password);
 		}
 		$this->sendRequest(
-			$url, $method, 'basic ' . \base64_encode($authString), false, $body
+			$url, $method, 'basic ' . \base64_encode($authString), false, $body, $header
 		);
 	}
 

--- a/tests/acceptance/features/bootstrap/OCSContext.php
+++ b/tests/acceptance/features/bootstrap/OCSContext.php
@@ -487,19 +487,26 @@ class OCSContext implements Context {
 	) {
 		$user = $this->featureContext->getActualUsername($user);
 		$ofUser = $this->featureContext->getActualUsername($ofUser);
-		$this->featureContext->verifyTableNodeColumns($table, ['endpoint']);
+		$this->featureContext->verifyTableNodeColumns($table, ['endpoint'], ['destination']);
 		$this->featureContext->emptyLastHTTPStatusCodesArray();
 		$this->featureContext->emptyLastOCSStatusCodesArray();
 		foreach ($table->getHash() as $row) {
 			$row['endpoint'] = $this->featureContext->substituteInLineCodes(
 				$row['endpoint'], $ofUser
 			);
+			$header = [];
+			if (isset($row['destination'])) {
+				$header['Destination'] = $this->featureContext->substituteInLineCodes(
+					$this->featureContext->getBaseUrl() . $row['destination'], $ofUser
+				);
+			}
 			$this->featureContext->authContext->userRequestsURLWithUsingBasicAuth(
 				$user,
 				$row['endpoint'],
 				$method,
 				$password,
-				$body
+				$body,
+				$header
 			);
 			$this->featureContext->pushToLastHttpStatusCodesArray(
 				$this->featureContext->getResponse()->getStatusCode()

--- a/tests/acceptance/features/bootstrap/WebDav.php
+++ b/tests/acceptance/features/bootstrap/WebDav.php
@@ -2025,6 +2025,25 @@ trait WebDav {
 	}
 
 	/**
+	 * @Then the HTTP status code of responses on all endpoints should be :statusCode1 or :statusCode2
+	 *
+	 * @param $statusCode1
+	 * @param $statusCode2
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function theHTTPStatusCodeOfResponsesOnAllEndpointsShouldBeOr($statusCode1, $statusCode2) {
+		$duplicateRemovedStatusCodes = \array_unique($this->lastHttpStatusCodesArray);
+		foreach ($duplicateRemovedStatusCodes as $status) {
+			$status = (string)$status;
+			if (($status != $statusCode1) && ($status != $statusCode2)) {
+				Assert::fail("Unexpected status code received " . $status);
+			}
+		}
+	}
+
+	/**
 	 * @Then the OCS status code of responses on all endpoints should be :statusCode
 	 *
 	 * @param $statusCode

